compiler_opt/tools/generate_default_trace.py

```diff
@@ -196,6 +205,11 @@ def generate_trace(worker_manager_class: type[
         f'succeeded, and {total_training_examples} trainining examples '
         'written')
 
+  if _KEYS_FILE.value is not None:
+    with open(_KEYS_FILE.value, 'w', encoding='utf-8') as keys_file:
+      for key in all_keys:
+        keys_file.write(f'{key}\n')
```

Comment:
Use a single `.write()` call instead of multiple calls.
```suggestion
      keys_file.write('\n'.join(str(key) for key in all_keys) + '\n')
```